### PR TITLE
Adds years tabs to winners page

### PR DESCRIPTION
**Pull request by bycdiaz**
Clicks on year tabs show winners based on year of prize.

### CODE REVIEW

File: public/javascript/gunDrawing.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const years = document.querySelector(".years");
+const cardDates = document.querySelectorAll(".date");
+
+function winnerDisplay() {
+  cardDates.forEach(date => {
+    const winnerYear = date.innerHTML.slice(-4);
```
Comment (I3uckwheat) [inline public/javascript/gunDrawing.js:6]: Shouldn't rely on `innerHTML` for data like this. Use something like data-date.

File: views/gunDrawing/index.pug [hunk 2]
```diff
@@ -18,18 +18,44 @@ block content
...
+          if user
+            .adminControls
+              button(type="button").edit edit
+              button(type="button").delete delete
+          h3.name= winner.fullName
+          h5.location= winner.location
+          p.gun= winner.gun
+          p.date= winner.date.toDateString()
+          .shortLine
+      else
```
Comment (I3uckwheat) [inline views/gunDrawing/index.pug:46]: There is a lot of duplication here. Having to change the template at all within a card, means one has to change it in both places. Why not have the `if` deciding what class is displayed only decide if it renders like so:

```pug
if winner.date.toDateString().slice(-4) === currentYear
  .card.current-year(id=winner._id)
else 
  .card.hidden(id=winner._id)
```

This way the template is kept smaller, as well as making it easier to edit.

File: .gitignore [hunk 3]
```diff
@@ -13,3 +13,6 @@ palette.pdf
 README.md
 public/javascript/src
 public/javascript/src/*
+grotonfiresandbox
+.env
+variables.env.sample
```
Comment (I3uckwheat) [inline .gitignore:18]: This should be committed to help remember what variables need to be populated for the code to run.

File: .gitignore [hunk 4]
```diff
@@ -13,3 +13,6 @@ palette.pdf
 README.md
 public/javascript/src
 public/javascript/src/*
+grotonfiresandbox
```
Comment (I3uckwheat) [inline .gitignore:16]: Do not need this.

File: controllers/gunDrawingController.js [hunk 5]
```diff
@@ -6,7 +6,15 @@ const Winner = mongoose.model('Winner');
 exports.index = (req, res) => {
   Winner.find().sort({ date: -1 })
     .then((winners) => {
-      res.render('gunDrawing/index', { winners, title: 'Gun-Drawing Winners' });
+      let allYears = []
+      const date = new Date();
+      const currentYear = date.getFullYear().toString();
+      
+      winners.forEach(winner => {
+        const winnerYear = Number(winner.date.toString().slice(11, 15));
```
Comment (I3uckwheat) [inline controllers/gunDrawingController.js:14]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getFullYear

File: public/javascript/gunDrawing.js (hunk 1 above)
Comment (bycdiaz) [reply public/javascript/gunDrawing.js:6]: Added data attribute for in template and grabbed it from this line.

File: views/gunDrawing/index.pug (hunk 2 above)
Comment (bycdiaz) [reply views/gunDrawing/index.pug:46]: This seems to be a limitation of Pug. See [this issue](https://github.com/pugjs/pug/issues/1779) regarding the topic. In the future, using something like Handlebars could be beneficial.

File: views/gunDrawing/index.pug [hunk 6]
```diff
@@ -18,18 +18,44 @@ block content
...
-        if user
-          .adminControls
-            button(type="button").edit edit
-            button(type="button").delete delete
-        h3.name= winner.fullName
-        h5.location= winner.location
-        p.gun= winner.gun
-        p.date= winner.date.toDateString()
-        .shortLine
+      if winner.date.toDateString().slice(-4) === currentYear
```
[the remark this answers is not in the document]
Comment (bycdiaz) [reply views/gunDrawing/index.pug:35]: Removed the `slice()` and replaced it with `getFullYear()`

File: .gitignore (hunk 3 above)
Comment (bycdiaz) [reply .gitignore:18]: Got it. Just removed `grotonfiresandbox` and `variables.env.sample` from the `.gitignore` file. `.env` will stay.

**Comment by bycdiaz** [comment]
Closed due to history conflicts. Opening new PR.

**Pull request closed without merging**